core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.controller.ts

```diff
@@ -108,34 +212,51 @@ angular.module('oppia').controller('TranslationModalController', [
     };
 
     $scope.suggestTranslatedText = function() {
-      if (!$scope.uploadingTranslation && !$scope.loadingData) {
-        SiteAnalyticsService.registerContributorDashboardSubmitSuggestionEvent(
-          'Translation');
-        $scope.uploadingTranslation = true;
-        var imagesData = ImageLocalStorageService.getStoredImagesData();
-        ImageLocalStorageService.flushStoredImagesData();
-        ContextService.resetImageSaveDestination();
-        TranslateTextService.suggestTranslatedText(
-          $scope.activeWrittenTranslation.html,
-          TranslationLanguageService.getActiveLanguageCode(),
-          imagesData, function() {
-            AlertsService.addSuccessMessage(
-              'Submitted translation for review.');
-            if ($scope.moreAvailable) {
-              var textAndAvailability = (
-                TranslateTextService.getTextToTranslate());
-              $scope.textToTranslate = textAndAvailability.text;
-              $scope.moreAvailable = textAndAvailability.more;
-            }
-            $scope.previousTranslationAvailable = true;
-            $scope.activeWrittenTranslation.html = '';
-            $scope.uploadingTranslation = false;
-          }, () => {
-            $uibModalInstance.close();
-          });
-      }
-      if (!$scope.moreAvailable) {
-        $uibModalInstance.close();
+      $scope.hasImgCopyError = false;
+      $scope.hasImgTextError = false;
+      const originalElements = angular.element(
+        $scope.textToTranslate);
+      const translatedElements = angular.element(
+        $scope.activeWrittenTranslation.html);
+
+      let translationError = $scope.validateImages(
+        originalElements, translatedElements);
+
+      if (translationError.hasUncopiedImgs) {
+        $scope.hasImgCopyError = true;
+      } else if (translationError.hasDuplicateAltTexts ||
+        translationError.hasDuplicateDescriptions) {
+        $scope.hasImgTextError = true;
+      } else {
+        if (!$scope.uploadingTranslation && !$scope.loadingData) {
```

Comment:
To reduce nesting, can we introduce a guard clause , i.e. invert this condition and break early?

```suggestion
        if ($scope.uploadingTranslation || $scope.loadingData) {
          break;
        }
```

Reply:
Done